src/OptimalTransport.jl

```diff
@@ -672,4 +675,162 @@ function quadreg(mu, nu, C, ϵ; θ=0.1, tol=1e-5, maxiter=50, κ=0.5, δ=1e-5)
+they are 1-Dimensional distributions and the cost
+function is of the form ``c(x,y) = h(|x-y|)`` such that
+``h`` is a convex function.
+
+Parameters:\n
+c: The cost function, which should be of the form ``c(x,y) = h(|x-y|)``
+where ``h`` is a convex function.
+μ: 1-D distribution (e.g. `μ = Distributions.Normal(0,1)`)\n
+ν: 1-D distribution (e.g. `μ = Distributions.Normal(0,1)`)\n
+
+Returns the optimal transport plan as a function
+  ```math
+  T(x)=F_\\nu^{-1}(F_\\mu(x)).
+  ```
+"""
+function optimal_transport_plan(
+    c, μ::ContinuousUnivariateDistribution, ν::UnivariateDistribution
+)
+    # Use T instead of γ to indicate that this is a Monge map.
+    T(x) = Distributions.quantile(ν, Distributions.cdf(μ, x))
+    return T
+end
+
+"""
+optimal_transport_cost(c, μ::DiscreteNonParametric, ν::DiscreteNonParametric; plan=nothing)
+
+Calculates the Optimal Transport Cost between μ to ν, where
+they are 1-Dimensional distributions and the cost
+function is of the form ``c(x,y) = h(|x-y|)`` such that
+``h`` is a convex function.
+
+Parameters:\n
+c: The cost function, which should be of the form ``c(x,y) = h(abs(x-y))``
+where ``h`` is a convex function. \n
+μ: Finite Discrete Distribution (e.g. `μ = DiscreteNonParametric(u, u_n)`)\n
+ν: Finite Discrete Distribution (e.g. `ν = DiscreteNonParametric(v, v_n)`)\n
+
+Returns the cost.
+"""
+function optimal_transport_cost(
+    c, μ::DiscreteNonParametric, ν::DiscreteNonParametric; plan=nothing
+)
+    if plan === nothing
+        return _optimal_transport_cost_plan(c, μ, ν)[1]
+    else
+        return dot(StatsBase.pairwise(c, μ.support, ν.support), plan)
+    end
+
+    return cost
+end
+
+"""
+optimal_transport_plan(c, μ::DiscreteNonParametric, ν::DiscreteNonParametric; plan=nothing)
+
+Calculates the Optimal Transport Plan between μ to ν, where
+they are 1-Dimensional distributions and the cost
+function is of the form ``c(x,y) = h(|x-y|)`` such that
+``h`` is a convex function.
+
+Parameters:\n
+c: The cost function, which should be of the form ``c(x,y) = h(abs(x-y))``
+where ``h`` is a convex function. \n
+μ: Finite Discrete Distribution (e.g. `μ = DiscreteNonParametric(u, u_n)`)\n
+ν: Finite Discrete Distribution (e.g. `ν = DiscreteNonParametric(v, v_n)`)\n
+
+Returns the optimal transport plan γ as a matrix.
+"""
+function optimal_transport_plan(
+    c, μ::DiscreteNonParametric, ν::DiscreteNonParametric; plan=nothing
+)
+    return _optimal_transport_cost_plan(c, μ, ν)[2]
+end
+
+"""
+function _optimal_transport_cost_plan(
+    c, μ::DiscreteNonParametric, ν::DiscreteNonParametric
+)
+
+Auxiliary function that calculates the cost and plan for
+μ an ν where both  they are 1-Dimensional discrete distributions with finite support
+and the cost function is of the form ``c(x,y) = h(|x-y|)`` such that
+``h`` is a convex function.
+
+Returns cost and γ, where cost represtents the optimal transport cost and
+γ is the optimal transport plan given as a matrix.
+"""
```

Comment:
It's not necessary to add docstrings for internal methods but it's great if there is some documentation :+1: However, as above the format should be fixed.